### DISCUSSION TITLE
Chunk server reporting 1TiB max of free space

### DISCUSSION THREAD

**Discussion by kaiyoti** [moosefs/moosefs#445] [Q&A]
Hi, I'm new to moosefs, and I just successfully mounted my hard drive as chunk server but I'm seeing that mfs info only shows that the disk have a total of 1TiB (when it actually has 3.6 TiB)

I tried different configurations... if I use -3.5TiB, it shows total size of 3.6TiB, this is the correct amount but obviously not what I want
If I use -1.5TiB, it shows total size of 2.5TiB. It seems like there's some limit of 1TiB free space? I tried looking the manual and couldn't find any info on this.

**Comment by chogata** [comment]
There are no limits for free space on a chunkserver. What operating system are you using on the chunkserver? What is the filesystem on the 3.6 TiB drive? Do you use compression in your filesystem?

**Reply by chogata** [reply]
Can you post here the output of `df -h` command on your operating system?

**Reply by kaiyoti** [reply]
Odroid Xu4 running Armbian, afaik, no compression (not entirely sure how to check this), it is exfat.
I compiled moosefs with default args:

_git clone https://github.com/moosefs/moosefs.git
cd moosefs
./linux_build.sh
sudo make install_

then ran:
_sudo sh moosefs/debian/moosefs-chunkserver.postinst configure
mkdir mfschunks
sudo chown mfs:mfs mfschunks`)

output of `df -h`:

_Filesystem      Size  Used Avail Use% Mounted on
udev            930M     0  930M   0% /dev
tmpfs           200M  6.6M  193M   4% /run
/dev/mmcblk0p1   29G  4.0G   25G  14% /
tmpfs           998M     0  998M   0% /dev/shm
tmpfs           5.0M  4.0K  5.0M   1% /run/lock
tmpfs           998M     0  998M   0% /sys/fs/cgroup
tmpfs           998M   12K  998M   1% /tmp
/dev/loop3      144M  144M     0 100% /snap/gnome-3-28-1804/146
/dev/loop2       66M   66M     0 100% /snap/gtk-common-themes/1515
/dev/loop1      120M  120M     0 100% /snap/chromium/1688
/dev/loop0      115M  115M     0 100% /snap/chromium/1525
/dev/loop5       27M   27M     0 100% /snap/snapd/12705
/dev/loop4       46M   46M     0 100% /snap/core18/2001
/dev/loop7      145M  145M     0 100% /snap/gnome-3-28-1804/163
/dev/loop6       46M   46M     0 100% /snap/core18/2072
/dev/loop8       65M   65M     0 100% /snap/gtk-common-themes/1514
/dev/loop9       27M   27M     0 100% /snap/snapd/11410
/dev/sda2       3.7T  147M  3.7T   1% /home/kaiyoti/mfschunks
/dev/zram0       49M  4.7M   41M  11% /var/log
tmpfs           200M   12K  200M   1% /run/user/1000
overlaid        200M   12K  200M   1% /run/user/1000/kaiyoti-firefox-dyq7djxm.default
overlaid        200M   12K  200M   1% /run/user/1000/kaiyoti-firefox-m4ht2nc9.default-release_


/dev/sda2 is the device

**Reply by kaiyoti** [reply]
well I converted it to GPT and now max is 2TiB instead of 1, still not quite there yet but it's better than before

**Reply by chogata** [reply]
It looks like some problem with statvfs on this particular Armbian. We've never encountered anything like that before (i.e. what MooseFS reports about a disk being different from what df reports).
I'm attaching a simple program that prints informations about a disk. Could you download it, compile it on your machine and run it with the disk's path (/dev/sda2) as parameter and post the output it prints?
Save it in any directory (your home directory for example) and run these commands when you are in this directory:
```
gunzip checkstatvfs.c.gz
make checkstatvfs
./checkstatvfs /dev/sda2

```
[checkstatvfs.c.gz](https://github.com/moosefs/moosefs/files/6923931/checkstatvfs.c.gz)

**Reply by kaiyoti** [reply]
_f_bsize:   4096
f_frsize:  4096
f_blocks:  237893
f_bfree:   237893
f_bavail:  237893
f_files:   186506
f_ffree:   185967
f_favail:  185967
f_fsid:    0
f_flag:    4106
f_namemax: 255
calc avail_all: 974409728
calc avail_user: 974409728
calc total_all: 974409728
calc total_user: 974409728_

**Reply by kaiyoti** [reply]
I followed this guide to repartition my drive in xfs format https://moosefs.com/blog/how-install-moosefs/
I'm getting the full 3.6TiB in total, it does show a 26GiB used from an empty state, is that normal?

**Reply by chogata** [reply]
Yes, this is normal, it's a small amount of reserved space for on-going operations. 
As for the previous results (for exfat, I assume) - the operating system is clearly reporting incorrect values via a system call - nothing we can do about it :(
